src/types/package.rs

```diff
@@ -125,6 +126,37 @@ impl FromStr for PackageType {
     }
 }
 
+impl From<PackageType> for purl::PackageType {
+    fn from(value: PackageType) -> Self {
+        match value {
+            PackageType::Npm => purl::PackageType::Npm,
+            PackageType::PyPi => purl::PackageType::PyPI,
+            PackageType::Maven => purl::PackageType::Maven,
+            PackageType::RubyGems => purl::PackageType::Gem,
+            PackageType::Nuget => purl::PackageType::NuGet,
+            PackageType::Cargo => purl::PackageType::Cargo,
+            PackageType::Golang => purl::PackageType::Golang,
+        }
+    }
+}
+
+impl TryFrom<&purl::PackageType> for PackageType {
+    type Error = purl::UnsupportedPackageType;
+
+    fn try_from(value: &purl::PackageType) -> Result<PackageType, purl::UnsupportedPackageType> {
```

Comment:
```suggestion
impl TryFrom<purl::PackageType> for PackageType {
    type Error = purl::UnsupportedPackageType;

    fn try_from(value: purl::PackageType) -> Result<PackageType, purl::UnsupportedPackageType> {
```

`PackageType` is a simple integer copy type so we don't need to take a reference to it here.